### PR TITLE
Refactor streamlist

### PR DESCRIPTION
**Pull request by jun-loke**
# General info

**Issue number**: N/A

**Task description**: 

- Refactor StreamList and friends so that there is a wrapper function that handles the onChange events
- Remove tests on StreamList that weren't testing Streamlist
- Rewrite tests for StreamList
- Write test for StreamsTableWrapper
- Write tests for StreamsTable
- 
**Update**: because the StreamList page's test file was testing the inner components and not the Streamlist, fhe file was rewritten to test only the Streamlist. However, this decreased the total code coverage from 95.1% to ~94.4% because this is a regression, additional tests and refactors will be added to this pr. However it's already proven to be a lot of work and a lot of new code. I kinda want to put it in its own pr
- Refactor delete stream Dialog & add tests
 
 
| Original   | Updated|
|:---------------:|:---------------:|
| | |

# Testing

**Test coverage**:

# Additional notes

**Note to reviewers**:

The Streamlist test was testing the StreamsTable component within it and not actually the StreamList. Those tests have been removed. 
**To do before merging**:
- [ ] Edit Changelog(if necessary)

### CODE REVIEW

File: frontend/src/streamlist/StreamsTablePopulator.jsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import React from "react";
-import { Box, Container } from "@material-ui/core";
 import PropTypes from "prop-types";
-import StreamsTable from "./StreamsTable";
+import ActiveStreamsTable from "./StreamsTable";
```
Comment (gdzooks) [inline frontend/src/streamlist/StreamsTablePopulator.jsx]: if there's only a single export can you forgo using `as` to change the names of stuff you import?
also could you clarify why we're renaming to `ActiveStreamsTable` ?

**Review by gdzooks — DISMISSED** [review]
short questions but the code looks & works fine for me :pray:

File: frontend/src/streamlist/StreamsTablePopulator.jsx (hunk 1 above)
Comment (jun-loke) [reply frontend/src/streamlist/StreamsTablePopulator.jsx]: typo. Removed in 965f203

**Comment by jun-loke** [comment]
Created issue #333 in order handle the refactor & adding back the tests of the DeleteStream Dialog

**Review by felixlapierre — DISMISSED** [review]
look good to me!

File: frontend/src/streamlist/__tests__/StreamListPage.test.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    wrapper = Enzyme.shallow(<StreamListPage dataSource={dummySource} />);
+  });
+  describe("Should contain the following components", () => {
+    it("Contains 1 <Container/> component", () => {
+      expect(wrapper.find(Container)).toHaveLength(1);
+    });
+    it("Contains 1 <Box/> component", () => {
+      expect(wrapper.find(Box)).toHaveLength(1);
+    });
+    it("Contains 1 <StreamTablePopulator/> component", () => {
```
Comment (gdzooks) [inline frontend/src/streamlist/__tests__/StreamListPage.test.jsx]: might want to update `<StreamTablePopulator/>` to `<StreamsTableWrapper/>`

**Review by gdzooks — DISMISSED** [review]
Tiny comment but otherwise lgtm!! hot DAMN that testing though :pray::heart:

File: frontend/src/streamlist/__tests__/StreamListPage.test.jsx (hunk 2 above)
Comment (jun-loke) [reply frontend/src/streamlist/__tests__/StreamListPage.test.jsx]: Fixed in 948a703

**Review by gdzooks — DISMISSED** [review]
:heart::tada:

**Review by gdzooks — APPROVED** [review]
:,3 :heart:

**Review by felixlapierre — APPROVED** [review]

**Pull request merged**